src/ui/Combobox/ComboboxValues.ts

```diff
@@ -1,32 +1,35 @@
-import { Combobox } from './Combobox';
+import { ICombobox, IComboboxValues, IComboboxValue } from './ICombobox';
 import { $$ } from '../../utils/Dom';
 import { find } from 'underscore';
 
-export interface IComboboxValue {
-  value: any;
-  element: HTMLElement;
-}
-
-export class ComboboxValues {
+export class ComboboxValues implements IComboboxValues {
   public element: HTMLElement;
   public mouseIsOverValue = false;
+  public isRenderingNewValues = false;
   private values: IComboboxValue[] = [];
   private keyboardActiveValue?: IComboboxValue;
+  private isScrollable: boolean;
+  private focusedValueId: string;
 
-  constructor(private combobox: Combobox) {
+  constructor(private combobox: ICombobox) {
     this.element = $$('ul', {
       id: `${this.combobox.id}-listbox`,
       role: 'listbox',
       className: 'coveo-combobox-values',
       ariaLabelledby: `${this.combobox.id}-input`
     }).el;
     $$(this.element).hide();
+
+    this.isScrollable = !!this.combobox.options.scrollable;
+    this.isScrollable && this.element.addEventListener('scroll', () => this.onScroll());
   }
 
   public renderFromResponse(response: any) {
+    this.isRenderingNewValues = true;
     this.clearValues();
     this.values = this.combobox.options.createValuesFromResponse(response);
     this.render();
+    this.isRenderingNewValues = false;
```

Comment:
Why is this flag needed?

Reply:
When we re-render, since the element is being emptied, it causes the blur to be called.